### PR TITLE
fix: Update provisioning model based on feedback

### LINKED ISSUE #7272: Provisioning modal : UI update
Provisioning modal

UI update to "Create new resources" within provisioning modal
- Includes better grouping
- updated copy for tooltips
- Link to learn more about Luis regions

Figma file here - https://www.figma.com/file/J3oIdXDmeBQbAe4a4PYbQc/Jaskeerat-s-files?node-id=5741%3A295

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Description

Updated the provisioning UI based on figma and change requests from the bug bash.

## Task Item

fixes #7272

## Screenshots

image
image
image

### CODE REVIEW

File: Composer/packages/client/src/pages/botProject/create-publish-profile/ProfileFormDialog.tsx [hunk 1]
```diff
@@ -70,6 +65,11 @@ export const ProfileFormDialog: React.FC<ProfileFormDialogProps> = (props) => {
     }
   };
 
+  const updateName = (e, newName) => {
+    setName(newName);
+    isValidateProfileName(newName);
```
Comment (hatpick) [inline Composer/packages/client/src/pages/botProject/create-publish-profile/ProfileFormDialog.tsx:70]: nit: name of the func -> ```validateProfileName```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 2]
```diff
@@ -75,27 +80,46 @@ const AddResourcesSectionName = styled(Text)`
   font-size: ${FluentTheme.fonts.mediumPlus.fontSize};
 `;
 
-const labelTooltipStyles = {
+const ConfigureResourcesSectionName = styled(Text)`
+  font-size: ${FluentTheme.fonts.mediumPlus.fontSize};
```
Comment (hatpick) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:84]: does it need px?

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 3]
```diff
@@ -75,27 +80,46 @@ const AddResourcesSectionName = styled(Text)`
...
+  font-weight: ${FontWeights.semibold};
+  margin-bottom: 4px;
+`;
+
+const ConfigureResourcesSectionDescription = styled(Text)`
+  font-size: ${FluentTheme.fonts.medium.fontSize};
```
Comment (hatpick) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:90]: same here

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 4]
```diff
@@ -787,91 +786,149 @@ export const AzureProvisionDialog: React.FC = () => {
...
-          options={luisLocationOptions}
-          placeholder={formatMessage('Select one')}
-          selectedKey={formData.luisLocation}
-          styles={{ root: { width: '75%' } }}
-          onChange={(e, o) => {
-            updateFormData('luisLocation', o.key as string);
-          }}
-        />
+        <Stack>
+          <ConfigureResourcesSectionName>Azure details</ConfigureResourcesSectionName>
```
Comment (hatpick) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:805]: formatMessage

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 5]
```diff
@@ -787,91 +786,149 @@ export const AzureProvisionDialog: React.FC = () => {
...
+          <ConfigureResourcesSectionDescription>
+            Select your Azure directory and subscription, enter resource group name.
```
Comment (hatpick) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:807]: formatMessage

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 6]
```diff
@@ -787,91 +786,149 @@ export const AzureProvisionDialog: React.FC = () => {
...
+              onChange={(choice) => {
+                setIsNewResourceGroup(choice.isNew);
+                updateFormData('resourceGroup', choice.name);
+                setErrorResourceGroupName(choice.errorMessage);
+              }}
+            />
+          </Stack>
+          <ConfigureResourcesSectionName>Resource details</ConfigureResourcesSectionName>
+          <ConfigureResourcesSectionDescription>
+            Enter resource name and select region. This will be applied to the new resources.
```
Comment (hatpick) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:875]: formatMessage

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 7]
```diff
@@ -787,91 +786,149 @@ export const AzureProvisionDialog: React.FC = () => {
...
+              resourceGroupNames={resourceGroupNames}
+              selectedResourceGroupName={isNewResourceGroup ? undefined : formData.resourceGroup}
+              onChange={(choice) => {
+                setIsNewResourceGroup(choice.isNew);
+                updateFormData('resourceGroup', choice.name);
+                setErrorResourceGroupName(choice.errorMessage);
+              }}
+            />
+          </Stack>
+          <ConfigureResourcesSectionName>Resource details</ConfigureResourcesSectionName>
```
Comment (hatpick) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:873]: formatMessage

File: Composer/packages/client/src/pages/botProject/create-publish-profile/ProfileFormDialog.tsx (hunk 1 above)
Comment (GeoffCoxMSFT) [reply Composer/packages/client/src/pages/botProject/create-publish-profile/ProfileFormDialog.tsx:70]: I'm just moving the function to be declared before it is used :)

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx (hunk 2 above)
Comment (GeoffCoxMSFT) [reply extensions/azurePublish/src/components/azureProvisionDialog.tsx:84]: Based on the figma.

Comment (hatpick) [reply extensions/azurePublish/src/components/azureProvisionDialog.tsx:84]: no, I mean this ${FluentTheme.fonts.mediumPlus.fontSize} will evaluate to '14' as a string with no px (I think)

Comment (hatpick) [reply extensions/azurePublish/src/components/azureProvisionDialog.tsx:84]: nvm, it seems like they are using px values as string,

**Review by benbrown — APPROVED** [review]

**Pull request merged**